domain/cosmos/tx/msg_simulator.go

```diff
@@ -90,13 +92,13 @@ func (c *txGasCalulator) BuildTx(
 		return nil, err
 	}
 
-	gasAdjusted, feecoin, err := c.PriceMsgs(ctx, txfeesClient, encodingConfig.TxConfig, account, chainID, msg...)
-	if err != nil {
-		return nil, err
+	priceInfo := c.PriceMsgs(ctx, encodingConfig.TxConfig, account, chainID, msg...)
+	if priceInfo.Err != "" {
```

Comment:
While `PriceInfo` abstraction might have some benefits, I am not sure why we changing error type to string, it seems like this abstraction encapsulates some client logic that requires this specific format? I think `txGasCalulator` should be client indepdendent abstraction.

Reply:
It was changed to string so that we can return this to the client as a JSON response.

Duplicating `err error` when a string error message is already present feels unnecessary.

Happy for this to be iterated in a follow-up if you're feeling strongly